src/compressAlignment.cpp

```diff
@@ -23,32 +23,45 @@ void shasta::compress(const Alignment& alignment, string& s)
         const int32_t skip0 = int32_t(ordinals[0]) - int32_t(ordinal0);
         const int32_t skip1 = int32_t(ordinals[1]) - int32_t(ordinal1);
 
+        ordinal0 = ordinals[0];
+        ordinal1 = ordinals[1];
+
         // Find the end of the streak.
-        uint32_t n = 0;
+        uint32_t n = 1;
         for(uint64_t j=i+1; j<alignment.ordinals.size(); j++, n++) {
-            ++ordinal0;
-            ++ordinal1;
             const auto& nextOrdinals = alignment.ordinals[j];
-            if(nextOrdinals[0] != ordinal0) {
+            if(nextOrdinals[0] != ordinal0 + 1) {
                 break;
             }
-            if(nextOrdinals[1] != ordinal1) {
+            if(nextOrdinals[1] != ordinal1 + 1) {
                 break;
             }
+            ++ordinal0;
+            ++ordinal1;
         }
 
+        i += n;
 
+        // cout << "(Skip0, Skip1, N) = (" << skip0 << ", " << skip1 << ", " << n << ")" << endl;
         // Serialize skipx, skipy, n into the output string
         // using the most compact format possible.
         if(Format0::ok(skip0, skip1, n)) {
             Format0 format(skip0, skip1, n);
             s.append(reinterpret_cast<char*>(&format), sizeof(format));
-
         } else if(Format1::ok(skip0, skip1, n)) {
             Format1 format(skip0, skip1, n);
             s.append(reinterpret_cast<char*>(&format), sizeof(format));
+        } else if(Format2::ok(skip0, skip1, n)) {
+            Format2 format(skip0, skip1, n);
+            s.append(reinterpret_cast<char*>(&format), sizeof(format));
+        } else if(Format3::ok(skip0, skip1, n)) {
+            Format3 format(skip0, skip1, n);
+            s.append(reinterpret_cast<char*>(&format), sizeof(format));
+        } else if(Format4::ok(skip0, skip1, n)) {
+            Format4 format(skip0, skip1, n);
+            s.append(reinterpret_cast<char*>(&format), sizeof(format));
         } else {
-            throw runtime_error("Unable to compress alignment streak: "
+            throw std::runtime_error("Unable to compress alignment streak: "
```

Comment:
This message would not be understandable by a user. Ok to leave as is, but add something to the effect "this is a Shasta internal error...". Or else use `SHASTA_ASSERT`.

Reply:
Good point. Will change to `SHASTA_ASSERT`.